packages/flutter_tools/lib/src/commands/drive.dart

```diff
@@ -314,8 +316,21 @@ $ex
       });
     }
 
+    List<String> experimentalFlags = <String>[];
+    if (argParser.options.containsKey(FlutterOptions.kEnableExperiment) &&
```

Comment:
You could use `getBuildInfo().dartExperiments` (or better, yet use an existing build info), which should already have this parsing logic baked in

Reply:
Ah - cool! I added `--sound-null-safety` too